tensorflow_serving/batching/batching_session.cc

```diff
@@ -340,19 +464,59 @@ Status BatchingSession::MergeInputTensors(
 
   // For each input tensor name, a vector of tensors from the individual tasks.
   std::map<string, std::vector<Tensor>> tensors_to_merge;
-
```

Comment:
I removed most of the padding code from MergeInputTensors.  Now I am only precomputing `max_dim_sizes` and call AddPadding (as you suggested) on tensor when it is about to be added to `tensors_to_merge`.